### PR TITLE
[BEAM-4201][SQL] Add integration test for PubsubIO JSON -> SQL

### PR DESCRIPTION
**Pull request by akedin**
Integration test for PubsubIO->SQL
------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `./gradlew build` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by akedin** [comment]
run java precommit

**Comment by akedin** [comment]
run java precommit

**Comment by akedin** [comment]
run java precommit

**Comment by akedin** [comment]
run java precommit

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsub.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+    DATETIME_FORMAT.printTo(topicName, Instant.now());
+
+    return topicName.toString() + "-" + String.valueOf(ThreadLocalRandom.current().nextLong());
+  }
+
+  /**
+   * Topic path where events will be published to.
+   */
+  public String evevntsTopicPath() {
```
Comment (kennknowles) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsub.java]: typo: evevnts

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsub.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+import org.joda.time.DateTime;
+import org.joda.time.Instant;
+import org.joda.time.format.DateTimeFormat;
+import org.joda.time.format.DateTimeFormatter;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * Test rule which creates a new topic with randomized name and exposed a method to publishto it.
```
Comment (kennknowles) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsub.java]: typo: publishto it

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsub.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+ *
+ * <p>Deletes topic on shutdown.
+ */
+public class TestPubsub implements TestRule {
```
Comment (kennknowles) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsub.java]: This is super cool

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+      this.successPredicate = successPredicate;
+      this.resultTopicPath = resultTopicPath;
+    }
+
+    @Override
+    public POutput expand(PCollection<? extends T> input) {
+      return
+          input
+              // assign a dummy key, need this to keep all events in the same state cell
+              .apply(WithKeys.of("dummyKey"))
```
Comment (kennknowles) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsub/TestPubsubSignal.java]: What about windowing? You may also want to `Window.into(GlobalWindow)`. I think also a good amount of this logic exists in the `PAssert` implementation.

**Comment by kennknowles** [comment]
(btw this is awesome framework buliding)

**Comment by akedin** [comment]
fixed typos, added global windowing

**Comment by akedin** [comment]
run python precommit

**Comment by akedin** [comment]
run python precommit

**Comment by akedin** [comment]
run python precommit

**Comment by kennknowles** [comment]
Python precommit is on its own for this one. If we had fancier infrastructure it wouldn't even have run.

**Comment by kennknowles** [comment]
I may be out of date - previously ITs were run only as postcommit. Gradle makes it more fluid.

**Comment by akedin** [comment]
This one is probably not wired up yet, I only added the targets to SQL build

**Comment by akedin** [comment]
+ @alanmyrvold @jasonkuster : I'm wiring up the Beam SQL Pubsub integration test to the main [javaPostCommit target](https://github.com/apache/beam/pull/5347/files#diff-c197962302397baf3a4cc36463dce5eaR148). I am not sure whether Jenkins job which runs postCommit has all permissions needed (create/delete Pubsub topics+subscriptions). Can you check or point me to the account being used?

**Comment by akedin** [comment]
run java postcommit

**Comment by akedin** [comment]
run java postcommit

**Comment by jasonkuster** [comment]
The service account being used should be an editor on the project; access shouldn't be a problem.

**Comment by akedin** [comment]
green

File: sdks/java/extensions/sql/build.gradle [hunk 5]
```diff
@@ -158,3 +160,26 @@ task runPojoExample(type: JavaExec) {
...
+  ])
+
+  include '**/*IT.class'
+  maxParallelForks 4
+  classpath = project(":beam-sdks-java-extensions-sql").sourceSets.test.runtimeClasspath
+  testClassesDirs = files(project(":beam-sdks-java-extensions-sql").sourceSets.test.output.classesDirs)
+  useJUnit { }
+}
+
+task postCommit {
```
Comment (kennknowles) [inline sdks/java/extensions/sql/build.gradle]: nit that is not your problem - I think mostly that modules should define suites and the top level should choose pre/post commit

**Pull request merged**